x-pack/plugins/viz_editor/public/datasource_plugin_registry.ts

```diff
@@ -17,7 +17,7 @@ export interface DatasourcePanelProps<S extends VisModel = VisModel> {
 
 export interface DatasourcePlugin<S extends VisModel = VisModel> {
   name: string;
-  toExpression?: (visModel: S, mode: 'view' | 'edit') => string;
+  toExpression: (visModel: S, mode?: 'preview' | 'full') => string;
```

Comment:
This can be used to later to reduce the data necessary to render the preview (e.g. sampling the data, restricting the range, ...)